src/levels.cpp

```diff
@@ -1024,13 +1024,29 @@ bool levLoadData(char const *name, Sha256 const *hash, char *pSaveName, GAME_TYP
 			return false;
 		}
 	}
-
+	// this will trigger upgrades 
 	if (!stageThreeInitialise())
 	{
 		debug(LOG_ERROR, "Failed stageThreeInitialise()!");
 		return false;
 	}
 
+	// droids were actually loaded before (during stageThreeInitialise) upgrades.
+	// So to actually apply them, we ll iterate over all droids, and re-calculate, once again,
+	// their upgradable parts.
+	// Without that, both Campaing and Skirmish saves are broken.
+	for (int player=0; player < MAX_PLAYERS; player++)
+	{
+			// one of these lists is empty when on mission
```

Comment:
Spelling errors in large comment: we'll, Campaign

And the comment in the loop has tabbing inconsistent with the rest of the code.